### PR TITLE
Fix: handle `tinyint(1)` BOOLEAN inference with modifiers like NOT NULL

### PR DESCRIPTION
**Pull request by WittgensteinBeetle**
### Problem

The `getFivetranDataType` function originally only returned `BOOLEAN` if the MySQL type was *exactly* `"tinyint(1)"`. This excluded common real-world variants like:

- `tinyint(1) NOT NULL`
- `tinyint(1) unsigned`
- `tinyint(1) zerofill`
- `TINYINT(1)` (case insensitivity)

These types should be safely treated as booleans when `treatTinyIntAsBoolean` is enabled — which matches MySQL connector behavior and expectations from most CDC systems.

---

### Fix

The logic has been updated to:
- Normalize `mysqlType` to lowercase
- Split it by whitespace (using `strings.Fields`)
- Only evaluate the **base type** (e.g., `"tinyint(1)"`) for the boolean match

This avoids false negatives while ensuring that `tinyint(2+)` are not misclassified as booleans.

---

### Test Coverage

Added unit tests for:

- `tinyint(1)` -- Pass 
- `tinyint(1) NOT NULL` --  Pass
- `tinyint(1) unsigned` -- Pass
- `tinyint(2)` -- Fail
- `tinyint` -- Fail
- `TINYINT(1)` -- Pass

Added an additional unit test to include these parameters as the original did not include these edge cases

### CODE REVIEW

File: cmd/internal/server/handlers/schema_builder_test.go [hunk 1]
```diff
@@ -256,3 +256,30 @@ func TestCanDetectDecimalPrecision(t *testing.T) {
 		})
 	}
 }
+
+//Adding an additional unit test for the tinyint(1) as bool handling since the orginals were not wide enough
```
Comment (maxenglander) [inline cmd/internal/server/handlers/schema_builder_test.go]: can remove this comment and the next line

File: cmd/internal/server/handlers/schema_builder_test.go [hunk 2]
```diff
@@ -256,3 +256,30 @@ func TestCanDetectDecimalPrecision(t *testing.T) {
...
+
+func TestGetFivetranDataType_TinyintBooleanHandling(t *testing.T) {
+	tests := []struct {
+		mysqlType             string
+		treatTinyIntAsBoolean bool
+		expected              fivetransdk.DataType
+	}{
+		{"tinyint(1)", true, fivetransdk.DataType_BOOLEAN},
```
Comment (maxenglander) [inline cmd/internal/server/handlers/schema_builder_test.go]: is there any reason these test cases can't fit as cases into the existing `TestSchema_CanPickRightFivetranType` test rather than defining a new test?

File: cmd/internal/server/handlers/schema_builder.go [hunk 3]
```diff
@@ -230,8 +230,20 @@ func parseEnumOrSetValues(mType string) ValueMap {
 // Convert columnType to fivetran type
 func getFivetranDataType(mType string, treatTinyIntAsBoolean bool) (fivetransdk.DataType, *fivetransdk.DecimalParams) {
 	mysqlType := strings.ToLower(mType)
+	// only interested in first part of the schema output to avoid issues with NOT NULL
+	parts := strings.Fields(mysqlType)
+
+	//empty schema condition
+	if len(parts) == 0 {
+		return fivetransdk.DataType_UNSPECIFIED, nil
```
Comment (maxenglander) [inline cmd/internal/server/handlers/schema_builder.go]: nice, can we add a test case for this?

**Review by maxenglander — COMMENTED** [review]
this looks good, will approve this once tests pass and my comments are addressed
